sample-core/src/commonMain/kotlin/com/mobilejazz/kmmsample/core/screen/mvi/hackerPostDetail/HackerPostDetailViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.mobilejazz.kmmsample.core.screen.mvi.hackerPostDetail
+
+import com.harmony.kotlin.application.ui.mvi.Action
+import com.harmony.kotlin.application.ui.mvi.ViewModel
+import com.harmony.kotlin.application.ui.mvi.ViewState
+import com.harmony.kotlin.common.logger.Logger
+import com.mobilejazz.kmmsample.core.feature.hackerposts.domain.interactor.GetHackerNewsPostInteractor
+import com.mobilejazz.kmmsample.core.feature.hackerposts.domain.model.HackerNewsPost
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+sealed class HackerPostDetailAction : Action {
+  object Refresh : HackerPostDetailAction()
+}
+
+sealed class HackerPostDetailViewState : ViewState {
+  object Loading : HackerPostDetailViewState()
+  class Error(val message: String) : HackerPostDetailViewState()
+  data class Content(val post: HackerNewsPost) : HackerPostDetailViewState()
+}
+
+class HackerPostDetailViewModel(
```

Comment:
I will postpone this decision until more people comment on it. 

A ViewState and an Action are our equivalents to the interfaces defined for us in the MVP. In there we were defining them in the same file. For me, it makes sense to have all in the same file because are really coupled with that implementation and also it is easy to just go to the top of the file and check the state or modify it if needed.